otx/algorithms/visual_prompting/tasks/inference.py

```diff
@@ -0,0 +1,282 @@
+                    SegmentAnything,
+                )
+
+                model = SegmentAnything(config=config, state_dict=state_dict)
+            else:
+                raise NotImplementedError(
+                    (f"Current selected model {config.model.name} is not implemented. " f"Use SAM instead.")
+                )
+            return model
+
+        if otx_model is None:
+            model = get_model(config=self.config)
+            logger.info(
+                "No trained model in project yet. Created new model with '%s'",
+                self.model_name,
+            )
+        else:
+            buffer = io.BytesIO(otx_model.get_data("weights.pth"))
+            model_data = torch.load(buffer, map_location=torch.device("cpu"))
+            from_lightning = False
+            if model_data.get("config", None):
+                from_lightning = True
+                logger.info("Load pytorch lightning checkpoint.")
+                if model_data["config"]["model"]["backbone"] != self.config["model"]["backbone"]:
+                    logger.warning(
+                        "Backbone of the model in the Task Environment is different from the one in the template. "
+                        f"creating model with backbone={model_data['config']['model']['backbone']}"
+                    )
+                    self.config["model"]["backbone"] = model_data["config"]["model"]["backbone"]
+            else:
+                logger.info("Load pytorch checkpoint.")
+
+            try:
+                model = get_model(config=self.config, state_dict=model_data["model"] if from_lightning else model_data)
+                logger.info("Complete to load model weights.")
+            except BaseException as exception:
+                raise ValueError("Could not load the saved model. The model file structure is invalid.") from exception
+
+        return model
+
+    def cancel_training(self) -> None:  # noqa: D102
+        raise NotImplementedError
+
+    def infer(self, dataset: DatasetEntity, inference_parameters: InferenceParameters) -> DatasetEntity:
+        """Perform inference on a dataset.
+
+        Args:
+            dataset (DatasetEntity): Dataset to infer.
+            inference_parameters (InferenceParameters): Inference parameters.
+
+        Returns:
+            DatasetEntity: Output dataset with predictions.
+        """
+        logger.info("Performing inference on the validation set using the base torch model.")
+        datamodule = OTXVisualPromptingDataModule(config=self.config.dataset, dataset=dataset)
+
+        logger.info("Inference Configs '%s'", self.config)
+
+        # Callbacks
+        inference_callback = InferenceCallback(otx_dataset=dataset)
+        callbacks = [TQDMProgressBar(), inference_callback]
+
+        self.trainer = Trainer(**self.config.trainer, logger=False, callbacks=callbacks)
+        self.trainer.predict(model=self.model, datamodule=datamodule)
```

Comment:
Just reminder, this should be changed to self.trainer.test

Reply:
Same as above.